integration/numpy_test.py

```diff
@@ -43,17 +42,89 @@
     print "Applied Cython patch"
 
 
-    subprocess.check_call([PYTHON_EXE, "setup.py", "install"], cwd=CYTHON_DIR)
-    subprocess.check_call([PYTHON_EXE, "-c", "import Cython"], cwd=CYTHON_DIR)
+    try:
+        subprocess.check_call([PYTHON_EXE, "setup.py", "install"], cwd=CYTHON_DIR)
+        subprocess.check_call([PYTHON_EXE, "-c", "import Cython"], cwd=CYTHON_DIR)
+    except:
+        subprocess.check_call(["rm", "-rf", CYTHON_DIR])
 else:
     print ">>> Cython already installed."
     print ">>>"
 
 print "\n>>>"
+print ">>> Patching NumPy..."
+print ">>>"
+NUMPY_PATCH_FILE = os.path.abspath(os.path.join(os.path.dirname(__file__), "numpy_patch.patch"))
+try:
+    cmd = ["patch", "-p1", "--forward", "-i", NUMPY_PATCH_FILE, "-d", NUMPY_DIR]
+    subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+except subprocess.CalledProcessError as e:
+    print e.output
+    if "Reversed (or previously applied) patch detected!  Skipping patch" not in e.output:
+        raise e
+
+print "\n>>>"
 print ">>> Setting up NumPy..."
 print ">>>"
 subprocess.check_call([PYTHON_EXE, "setup.py", "build"], cwd=NUMPY_DIR)
+
+print "\n>>>"
+print ">>> Installing NumPy..."
+print ">>>"
 subprocess.check_call([PYTHON_EXE, "setup.py", "install"], cwd=NUMPY_DIR)
 
+# From Wikipedia
+script = """
+import numpy as np
+from numpy.linalg import solve, inv
+a = np.array([[1, 2, 3], [3, 4, 6.7], [5, 9.0, 5]])
+b = np.array([3, 2, 1])
+a.transpose()
+inv(a)
+solve(a, b)
+print a, b
+"""
+
+# http://wiki.scipy.org/Tentative_NumPy_Tutorial/Mandelbrot_Set_Example
+mandelbrot = """
+from numpy import *
+
+def mandelbrot( h,w, maxit=20 ):
+        '''Returns an image of the Mandelbrot fractal of size (h,w).
+        '''
+        y,x = ogrid[ -1.4:1.4:h*1j, -2:0.8:w*1j ]
+        c = x+y*1j
+        z = c
+        divtime = maxit + zeros(z.shape, dtype=int)
+
+        for i in xrange(maxit):
+                z  = z**2 + c
+                diverge = z*conj(z) > 2**2
+                div_now = diverge & (divtime==maxit)
+                divtime[div_now] = i
+                z[diverge] = 2
+
+        return divtime
+
+m = mandelbrot(complex(400),complex(400))
+print m
```

Comment:
It doesn't return the right answer at the moment, still need some of those fixes on complex numbers. Which is why I left it as-is for now.